cunumeric/array.py

```diff
@@ -3019,6 +3019,17 @@ def ravel(self, order="C") -> ndarray:
         Multiple GPUs, Multiple CPUs
 
         """
+        if self.size == 0 and self.ndim > 1:
+            idx = tuple()
+            for i in range(self.ndim):
+                if self.shape[i] != 0:
+                    idx += (0,)
+                else:
+                    idx += (slice(None),)
+            idx = tuple(idx)
+            self = self[idx].copy()
+            return self
+
         return self.reshape(-1, order=order)
```

Comment:
I think just falling back to `self.reshape(-1, order=order)` works, and the special-case isn't necessary.

Reply:
I have added the test case that wouldn't work with the original implementation and moved this special case to `reshape`